### DISCUSSION TITLE
Overriding "Mode" colors (and weird gray thing)

### DISCUSSION THREAD

**Discussion by TheBabu** [NvChad/NvChad#2801] [Q&A] [answered]
Hello,
I was wondering if it was possible to override the mode colors on the bottom left, namely the "Normal", "Insert", and "Visual" mode colors.
![image](https://github.com/NvChad/NvChad/assets/24832107/f181f0e6-ff2b-4d99-aad7-e47f5f05ba64)

Additionally, I don't really like the grey space thing between the mode indicator and the file name :frowning_face: I was wondering if I could change the color of it such that it matches the color of the statusbar. 

Thanks for any help in advance!

**Comment by siduck** [comment]
which theme are you using

**Comment by TheBabu** [comment]
I am using my own custom base46 theme:
https://github.com/TheBabu/Dotfiles/blob/3a7063f0dbabffdcd0e29108a2390c37850b5187/.config/nvim/lua/themes/neodark.lua#L3

I am using the default statusline theme with block seperators:
https://github.com/TheBabu/Dotfiles/blob/3a7063f0dbabffdcd0e29108a2390c37850b5187/.config/nvim/lua/chadrc.lua#L7

**Comment by siduck** [comment]
override the statusline for this,  or make your grey color darker. Your theme looks like a mess cuz you dont seem to follow our base46 guidelines and many colors are so light

**Comment by TheBabu** [comment]
How would I go about overriding the statusline? Is there a direct variable that I can override?

**Reply by siduck** [reply]
read https://nvchad.com/docs/config/nvchad_ui

**Comment by TheBabu** [comment, marked as answer]
For those confused on how to do what I wanted to do:

To change the "Mode" colors one needs to use the `hl_override` option in `M.ui` in the `chadrc.lua` file. And to remove the weird gray spacer bar, I rewrote the `mode` function from https://github.com/NvChad/ui/blob/v2.5/lua/nvchad/stl/default.lua and also added it to my config.
```lua
M.ui = {
    ---@diagnostic disable-next-line
    theme = "neodark",
    statusline = {
        separator_style = "block",
        modules = {
            mode = function()
                if not utils.is_activewin() then
                    return ""
                end

                local modes = utils.modes

                local m = vim.api.nvim_get_mode().mode

                local current_mode = "%#St_" .. modes[m][2] .. "Mode#  " .. modes[m][1]
                local mode_sep1 = "%#St_" .. modes[m][2] .. "ModeSep#" .. sep_r
                return current_mode .. mode_sep1
            end,
...SNIP...
    hl_override = {
        St_NormalMode = { bg = "green" },
        St_NormalModeSep = { fg = "green" },
        St_InsertMode = { bg = "blue" },
        St_InsertModeSep = { fg = "blue" },
        St_VisualMode = { bg = "purple" },
        St_VisualModeSep = { fg = "purple" },
    }
```

**Reply by carlisia** [reply]
THANK YOU SO MUCH FOR THIS!! ‼️
